docs/en/changes/changes.md

```diff
@@ -50,6 +50,7 @@
 * Apply MQE on Virtual-Cache layer UI-templates
 * Add Echo component ID(5015) language: Golang.
 * Fix `index out of bounds exception` in `aggregate_labels` MQE function.
+* Fix Elasticsearch storage TTL timer, move created the latest index before retrieval indexes by aliases to avoid the 404 exception. 
```

Comment:
```suggestion
* Move created the latest index before retrieval indexes by aliases to avoid the 404 exception. This just prevents some interference from manual operations.
```

Reply:
First of all, this is unexpected from us that users manually change indices.

Then, @kezhenxu94 retrieve indices by alias without index could cause 404, is this an expected ES client behavior?

Reply:
And another possible cause: TTL deletes all indexes and fails to create the latest index (no chance to create again)

Reply:
> And another possible cause: TTL deletes all indexes and fails to create the latest index (no chance to create again)

In what situation TTL can delete all indices but fail to create new index? TTL delete only one outdated index per day, if all the indices are deleted it might indicate that it always fails when creating index and thus indicates the ES server is broken, why it succeeds to create new index if you move the creation to earlier?

Reply:
> > And another possible cause: TTL deletes all indexes and fails to create the latest index (no chance to create again)
> 
> In what situation TTL can delete all indices but fail to create new index? TTL delete only one outdated index per day, if all the indices are deleted it might indicate that it always fails when creating index and thus indicates the ES server is broken, why it succeeds to create new index if you move the creation to earlier?

yes, it just exists in theory, network or other issues that cause success to delete but fails to create.

I move create index earlier to give a chance to recreate index.